pkg/storage/export/export/export.go

```diff
@@ -417,52 +530,82 @@ func (ctrl *VMExportController) updateVMExport(vmExport *exportv1.VirtualMachine
 
 	if ctrl.isSourcePvc(&vmExport.Spec) {
 		return ctrl.handleIsSourcePvc(vmExport, service)
-	} else if ctrl.isSourceVMSnapshot(&vmExport.Spec) {
+	}
+	if ctrl.isSourceVMSnapshot(&vmExport.Spec) {
 		return ctrl.handleIsVmSnapshot(vmExport, service)
 	}
+	if ctrl.isSourceVM(&vmExport.Spec) {
+		return ctrl.handleIsVM(vmExport, service)
+	}
 	return 0, nil
 }
 
+func (ctrl *VMExportController) createPodAndSecret(vmExport *exportv1.VirtualMachineExport, pvcs []*corev1.PersistentVolumeClaim) (*corev1.Pod, error) {
+	if len(pvcs) == 0 {
+		// No PVCs
+		return nil, nil
+	}
+	pod, err := ctrl.createExporterPod(vmExport, pvcs)
+	if err != nil {
+		return nil, err
+	}
+	if pod == nil {
+		return nil, nil
+	}
+
+	if err := ctrl.getOrCreateCertSecret(vmExport, pod); err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
+
 func (ctrl *VMExportController) handleIsSourcePvc(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
-	pvcs := make([]*corev1.PersistentVolumeClaim, 0)
+	var pvcs []*corev1.PersistentVolumeClaim
+	isPopulated := false
 	pvc, exists, err := ctrl.getPvc(vmExport.Namespace, vmExport.Spec.Source.Name)
 	if err != nil {
 		return 0, err
-	} else if exists {
+	}
+	if exists {
 		pvcs = append(pvcs, pvc)
+		isPopulated, err = ctrl.isPVCPopulated(pvc)
+		if err != nil {
+			return 0, err
+		}
 	}
-
 	pod, exists, err := ctrl.getExporterPod(vmExport)
-	inUse := false
-	isPopulated := false
 	if err != nil {
 		return 0, err
-	} else if !exists {
+	}
+	inUse := false
+	if !exists {
 		inUse, err = ctrl.isPVCInUse(vmExport, pvc)
 		if err != nil {
 			return 0, err
 		}
 		if !inUse && len(pvcs) > 0 {
-			isPopulated, err = ctrl.isPVCPopulated(pvc)
-			if err != nil {
-				return 0, err
-			} else if isPopulated {
-				pod, err = ctrl.createExporterPod(vmExport, pvcs)
+			if isPopulated {
+				pod, err = ctrl.createPodAndSecret(vmExport, pvcs)
 				if err != nil {
 					return 0, err
-				} else if pod == nil {
-					return 0, nil
-				}
-
-				if err := ctrl.getOrCreateCertSecret(vmExport, pod); err != nil {
-					return 0, err
 				}
 			}
 		}
 	} else {
-		if err := ctrl.handlePodSucceededOrFailed(vmExport, pod); err != nil {
+		inUse, err = ctrl.isPVCInUse(vmExport, pvc)
```

Comment:
`isPVCInUse` is called first thing in both branches of this if statement.  So maybe do it before?  Also, if not `isPopulated` maybe infer inUse and not do a bunch of stuff here.  Maybe could clean up some of the conditional branches

Reply:
Sure let me see if I can clean that up.